### PR TITLE
update simulation utilities

### PR DESCRIPTION
**Pull request by edwinnglabs**
this PR just separate the logic of trend, seasonality and regression simulation.

### CODE REVIEW

File: orbit/utils/simulation.py [hunk 1]
```diff
@@ -5,14 +5,13 @@
...
-
     Parameters
     ----------
     series_len: int
         total length of series
-    type: str ['arma', 'rw']
+    method: str ['arma', 'rw']
     rw_loc: float
         mean of random walk
     rw_scale: float
```
Comment (steveyang90) [inline orbit/utils/simulation.py:17]: can you add a descriptive definition for this arg

File: orbit/utils/simulation.py [hunk 2]
```diff
@@ -53,14 +54,14 @@ def make_seasonality(series_len, seasonality, order=3, duration=1, scale=.05, ty
         in non-fourier type of seasonality ONLY.
     scale: float
         scale parameter of seasonality generation
-    type: str ['discrete', 'fourier']
+    method: str ['discrete', 'fourier']
```
Comment (steveyang90) [inline orbit/utils/simulation.py:57]: more descriptive docstrings

File: orbit/utils/simulation.py [hunk 3]
```diff
@@ -53,14 +54,14 @@ def make_seasonality(series_len, seasonality, order=3, duration=1, scale=.05, ty
...
     seed: int
     Returns
     -------
       np.array-llike with length equals `series_len`
```
Comment (steveyang90) [inline orbit/utils/simulation.py:61]: Please follow numpy docstring style guide for proper API docs parsing with sphinx https://numpydoc.readthedocs.io/en/latest/format.html

File: orbit/utils/simulation.py [hunk 4]
```diff
@@ -86,14 +87,73 @@ def make_seasonality(series_len, seasonality, order=3, duration=1, scale=.05, ty
...
+
 def make_ts_multiplicative(series_len=200, seasonality=-1, coefs=None, regressor_relevance=0.0,
                            regressor_log_loc=0.0, regressor_log_scale=0.2,
                            regressor_log_cov=None,
                            noise_to_signal_ratio=1.0, regression_sparsity=0.5,
                            obs_val_base=1000, regresspr_val_base=1000,
                            trend_type='rw', rw_loc=0.001, rw_scale=0.1, seas_scale=.05,
                            response_col='y', seed=0):
-    """
+    """ [Deprecated] Module to generate multiplicative time-series with trend, seasonality and regression components
```
Comment (steveyang90) [inline orbit/utils/simulation.py:156]: we should update all instances of this function being used in tutorial notebooks

**Review by steveyang90 — CHANGES_REQUESTED** [review]

File: orbit/utils/simulation.py (hunk 4 above)
Comment (edwinnglabs) [reply orbit/utils/simulation.py:156]: They are not being used right in the current tutorials.  Once the other decomposed modules updated, I may bring back a demo using those new modules @steveyang90

**Comment by edwinnglabs** [comment]
- [ ] can you add a descriptive definition for arg `method`
- [ ] follow numpy docstring style guide for proper API docs parsing with sphinx https://numpydoc.readthedocs.io/en/latest/format.html

File: orbit/utils/simulation.py [hunk 5]
```diff
@@ -23,14 +22,16 @@ def make_trend(series_len, rw_loc=0.001, rw_scale=0.1, type='rw', seed=1):
       np.array-llike with length equals `series_len`
     """
     # make trend
-    if type == "rw":
+    if method == "rw":
         rw = np.random.default_rng(seed).normal(rw_loc, rw_scale, series_len)
         trend = np.cumsum(rw)
-    elif type == "arma":
+    elif method == "arma1":
```
[the remark this answers is not in the document]
Comment (edwinnglabs) [reply orbit/utils/simulation.py]: yeah but it's okay i guess let's use arma for tag.  it is enforcing arma(1) down stream but i feel like 'arma1' is hard to remember.

File: orbit/utils/simulation.py [hunk 6]
```diff
@@ -5,14 +5,13 @@
...
-def make_trend(series_len, rw_loc=0.001, rw_scale=0.1, type='rw', seed=1):
+def make_trend(series_len, rw_loc=0.0, rw_scale=0.1, method='rw', seed=1):
     """ Module to generate time-series trend with different methods
-
     Parameters
     ----------
     series_len: int
         total length of series
-    type: str ['arma', 'rw']
+    method: str ['arma', 'rw']
```
[the remark this answers is not in the document]
Comment (edwinnglabs) [reply orbit/utils/simulation.py:14]: added

File: orbit/utils/simulation.py (hunk 1 above)
Comment (edwinnglabs) [reply orbit/utils/simulation.py:21]: added

File: orbit/utils/simulation.py (hunk 2 above)
Comment (edwinnglabs) [reply orbit/utils/simulation.py]: added

**Comment by edwinnglabs** [comment]
should be good for another review @steveyang90 
here is the [notebook](https://nbviewer.jupyter.org/github/uber/orbit/blob/simulation-decomp/examples/simulation.ipynb)

**Review by steveyang90 — APPROVED** [review]
Changes look good!

**Pull request merged**